### PR TITLE
puller(cdc): refactor the puller

### LINKED ISSUE #6257: refactoring cdc puller
as title

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #6257 

### What is changed and how it works?

* move the `DDLPuller` from `owner` package to the `puller` package
* puller use standard context as parameters 

#### Tests 

 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by sdojjy — APPROVED** [review]

File: cdc/processor/processor.go [hunk 1]
```diff
@@ -739,23 +739,22 @@ func (p *processor) handleErrorCh(ctx cdcContext.Context) error {
...
 func (p *processor) createAndDriveSchemaStorage(ctx cdcContext.Context) (entry.SchemaStorage, error) {
 	kvStorage := p.upstream.KVStorage
-	ddlspans := []regionspan.Span{regionspan.GetDDLSpan(), regionspan.GetAddIndexDDLSpan()}
 	checkpointTs := p.changefeed.Info.GetCheckpointTs(p.changefeed.Status)
 	kvCfg := config.GetGlobalServerConfig().KVClient
 	stdCtx := contextutil.PutTableInfoInCtx(ctx, -1, puller.DDLPullerTableName)
 	stdCtx = contextutil.PutChangefeedIDInCtx(stdCtx, ctx.ChangefeedVars().ID)
 	stdCtx = contextutil.PutRoleInCtx(stdCtx, util.RoleProcessor)
-	ddlPuller := puller.NewPuller(
+	ddlPuller := puller.New(
```
Comment (asddongmen) [inline cdc/processor/processor.go:747]: Should it be `NewDDLPuller` here?

Comment (3AceShowHand) [reply cdc/processor/processor.go:747]: not, this `ddlPuller` has such a name, but does not implement the `DDLPuller` intrerface.

File: cdc/puller/ddl_puller_test.go [hunk 2]
```diff
@@ -267,7 +269,7 @@ func TestResolvedTsStuck(t *testing.T) {
 	waitResolvedTsGrowing(t, p, 30)
 	require.Equal(t, logs.Len(), 0)
 
-	mockClock.Add(2 * ownerDDLPullerStuckWarnTimeout)
+	mockClock.Add(2 * ddlPullerStuckWarnTimeout)
```
Comment (asddongmen) [inline cdc/puller/ddl_puller_test.go:272]: Please use `ddlPullerStuckWarnDuration` as other similar constant in this repo.

**Review by asddongmen — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**